ambry-router/src/main/java/com.github.ambry.router/GetBlobInfoOperation.java

```diff
@@ -0,0 +1,243 @@
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.TreeMap;
+
+
+/**
+ * GetBlobInfoOperation class is responsible for maintaining the state associated with a getBlobInfo operation,
+ * and completing it. A GetBlobInfo operation only needs to make requests for a single chunk to get the BlobInfo -
+ * which is either the only chunk in the case of a simple blob, or the metadata chunk in the case of composite blobs.
+ */
+class GetBlobInfoOperation extends GetOperation<BlobInfo> {
+  private BlobInfo blobInfo;
+  private final SimpleOperationTracker operationTracker;
+  // map of correlation id to the request metadata for every request issued for this operation.
+  protected final Map<Integer, GetRequestInfo> correlationIdToGetRequestInfo = new TreeMap<Integer, GetRequestInfo>();
+
+  /**
+   * Construct a GetBlobInfoOperation
+   * @param routerConfig the {@link RouterConfig} containing the configs for get operations.
+   * @param clusterMap the {@link ClusterMap} of the cluster
+   * @param responseHandler the {@link ResponseHandler} responsible for failure detection.
+   * @param blobIdStr the blob id associated with the operation in string form.
+   * @param futureResult the future that will contain the result of the operation.
+   * @param callback the callback that is to be called when the operation completes.
+   * @param time the Time instance to use.
+   * @throws RouterException if there is an error with any of the parameters, such as an invalid blob id.
+   */
+  GetBlobInfoOperation(RouterConfig routerConfig, ClusterMap clusterMap, ResponseHandler responseHandler,
+      String blobIdStr, FutureResult<BlobInfo> futureResult, Callback<BlobInfo> callback, Time time)
+      throws RouterException {
+    super(routerConfig, clusterMap, responseHandler, blobIdStr, futureResult, callback, time);
+    operationTracker = new SimpleOperationTracker(routerConfig.routerDatacenterName, blobId.getPartition(), true,
+        routerConfig.routerGetSuccessTarget, routerConfig.routerGetRequestParallelism);
+  }
+
+  /**
+   * Return the {@link MessageFormatFlags} to associate with a getBlobInfo operation.
+   * @return {@link MessageFormatFlags#BlobInfo}
+   */
+  @Override
+  MessageFormatFlags getOperationFlag() {
+    return MessageFormatFlags.BlobInfo;
+  }
+
+  /**
+   * For this operation, create and populate get requests (in the form of {@link RequestInfo}) to send out.
+   * @param requestRegistrationCallback the {@link RequestRegistrationCallback} to call for every request that gets
+   *                                    created as part of this poll operation.
+   */
+  @Override
+  void poll(RequestRegistrationCallback<GetOperation> requestRegistrationCallback) {
+    //First, check if any of the existing requests have timed out.
+    Iterator<Map.Entry<Integer, GetRequestInfo>> inFlightRequestsIterator =
```

Comment:
can we move this to a separate method similar to deletes. (cleanUpExpired/TimeOutRequests)